x-pack/plugins/security_solution/public/plugin.tsx

```diff
@@ -217,6 +225,65 @@ export class Plugin implements IPlugin<PluginSetup, PluginStart, SetupPlugins, S
     this.services.stop();
   }
 
+  public async registerDiscoverSharedFeatures(
+    core: CoreSetup<StartPluginsDependencies, PluginStart>,
+    plugins: SetupPlugins
+  ) {
+    const { discoverShared } = plugins;
+    const discoverFeatureRegistry = discoverShared.features.registry;
+    const cellRendererFeature: SecuritySolutionCellRendererFeature = {
+      id: 'security-solution-cell-renderer',
+      getRenderer: async () => {
+        const { getCellRendererForGivenRecord } = await this.getLazyDiscoverSharedDeps();
+        return getCellRendererForGivenRecord;
+      },
+    };
+
+    const appWrapperFeature: SecuritySolutionAppWrapperFeature = {
+      id: 'security-solution-app-wrapper',
+      getWrapper: async () => {
+        const [coreStart, startPlugins] = await core.getStartServices();
+
+        const services = await this.services.generateServices(coreStart, startPlugins);
+
+        const { createSecuritySolutionDiscoverAppWrapperGetter } =
+          await this.getLazyDiscoverSharedDeps();
+
+        return createSecuritySolutionDiscoverAppWrapperGetter({
+          core: coreStart,
+          services,
+          plugins: startPlugins,
+        });
+      },
```

Comment:
Do we need to provide Security cellActions in the Flyout when rendered in Discover? I miss a call to `this.registerActions` somewhere

Reply:
As mentioned in PR description, https://github.com/elastic/kibana/issues/196667 tackles actions and will be done separately.

We do not want to pass security cell actions such as `Filter in` , `Filter out` directly because Security's `Filter In` is different from Discover's `Filter In`.

Instead we want to take instances of those actions from Discover and then use them whenever a security component in running in Discover.  We still need to see how to do that.